### ISSUE TITLE
TypeError: Cannot read property 'getAssets' of undefined

### ISSUE THREAD

**Issue by ractoon** [stidges/laravel-mix-mjml#12]
Running into an issue with `mix` and `mix watch` when this plugin is included, details below:

```
/node_modules/webpack/lib/Compilation.js:4633
                                                                if (!inTry) throw err;
                                                                            ^

TypeError: Cannot read property 'getAssets' of undefined
    at /node_modules/laravel-mix/node_modules/mini-css-extract-plugin/dist/loader.js:290:37
    at /node_modules/webpack/lib/Compiler.js:549:20
    at /node_modules/webpack/lib/Compiler.js:1176:25
    at finalCallback (/node_modules/webpack/lib/Compilation.js:2785:11)
    at /node_modules/webpack/lib/Compilation.js:3057:18
    at eval (eval at create (/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:20:1)
    at fn (/node_modules/webpack/lib/Compilation.js:479:17)
    at Hook.eval [as callAsync] (eval at create (/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:18:1)
    at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (/node_modules/webpack/node_modules/tapable/lib/Hook.js:18:14)
    at cont (/node_modules/webpack/lib/Compilation.js:3055:34)
    at /node_modules/webpack/lib/Compilation.js:3103:10
    at symbolIterator (/node_modules/neo-async/async.js:3485:9)
    at done (/node_modules/neo-async/async.js:3527:9)
    at /node_modules/neo-async/async.js:2830:7
    at done (/node_modules/neo-async/async.js:2865:11)
    at /node_modules/neo-async/async.js:2818:7
    at /node_modules/webpack/lib/Compilation.js:4626:18
    at /node_modules/webpack/lib/HookWebpackError.js:68:3
    at Hook.eval [as callAsync] (eval at create (/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:13:1)
    at Cache.store (/node_modules/webpack/lib/Cache.js:107:20)
    at ItemCacheFacade.store (/node_modules/webpack/lib/CacheFacade.js:137:15)
    at /node_modules/webpack/lib/Compilation.js:4623:25
    at /node_modules/webpack/lib/Cache.js:91:34
    at Array.<anonymous> (/node_modules/webpack/lib/cache/MemoryCachePlugin.js:45:13)
    at /node_modules/webpack/lib/Cache.js:91:19
    at Hook.eval [as callAsync] (eval at create (/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:17:1)
    at Cache.get (/node_modules/webpack/lib/Cache.js:75:18)
    at ItemCacheFacade.get (/node_modules/webpack/lib/CacheFacade.js:111:15)
    at /node_modules/webpack/lib/Compilation.js:4532:22
    at arrayEach (/node_modules/neo-async/async.js:2405:9)
    at Object.each (/node_modules/neo-async/async.js:2846:9)
    at /node_modules/webpack/lib/Compilation.js:4521:14
    at symbolIterator (/node_modules/neo-async/async.js:3482:9)
    at timesSync (/node_modules/neo-async/async.js:2297:7)
    at Object.eachLimit (/node_modules/neo-async/async.js:3463:5)
    at Compilation.createChunkAssets (/node_modules/webpack/lib/Compilation.js:4498:12)
```

package.json:

```
 "devDependencies": {
   ...
    "laravel-mix": "^6.0.19",
    "laravel-mix-mjml": "^4.0.0",
    "laravel-mix-polyfill": "^3.0.1",
    "targets-webpack-plugin": "^3.0.0",
    "webpack": "^5.23.0"
  },
```

Any pointers would be greatly appreciated. Not sure if there's something I can add to config to prevent this.

**Comment by ractoon** [comment]
Disregard, user error. Wrapping the `const response = mjml2html(fs.readFileSync(entry, 'utf8'), mjmlOptions);` call in a try/catch revealed that the library was encountering a malformed mjml error (missing wrapping mjml/mjml-body on components) that was causing the issue. Once those tags were in place in the components everything worked as expected.

**Comment by stidges** [comment]
Thank you for reporting this. I added a try catch to the compilation to prevent this from crashing your build process! It's available now under version v4.0.1 :)
